pyproject.toml

```diff
@@ -40,6 +40,7 @@ dependencies = [
     "typing_extensions",
     "eval_type_backport",
     "polars>=0.20.22",
+    "skmultilearn",
```

Comment:
Have you had a look at how this iterative subsampling function works in skmultilearn? If we can implement it ourselves, I would probably prefer that over introducing a new dependency. If it would be too complex to implement in MTEB we can of course roll with sklmultilearn.

Reply:
Yea, I had a quick look. It's not terribly complicated but there's a few non-trivial things going on, see:
- https://github.com/scikit-multilearn/scikit-multilearn/blob/master/skmultilearn/model_selection/iterative_stratification.py#L175
- ["A Network Perspective on Stratification of Multi-Label Data", Szymański & Kajdanowicz, 2017](http://proceedings.mlr.press/v74/szyma%C5%84ski17a/szyma%C5%84ski17a.pdf)